src/MICore/LaunchOptions.cs

```diff
@@ -172,6 +172,63 @@ public EnvironmentEntry(Xml.LaunchOptions.EnvironmentEntry xmlEntry)
         public string Value { get; private set; }
     }
 
+    public sealed class SourceMapEntry
+    {
+        public SourceMapEntry() // used by launchers 
+        {
+        }
+
+        public SourceMapEntry(Xml.LaunchOptions.SourceMapEntry xmlEntry)
+        {
+            this.EditorPath = xmlEntry.EditorPath;
+            this.CompileTimePath = xmlEntry.CompileTimePath;
+            this.UseForBreakpoints = xmlEntry.UseForBreakpoints;
+        }
+
+        private string _editorPath;
+        public string EditorPath
+        {
+            get
+            {
+                return _editorPath;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("EditorPath");
+                }
+                this._editorPath = value;
```

Comment:
We might want to reject the empty string also, as currently this will cause us to crash (and clearly it shouldn't be used anyway).